Library/Homebrew/cask/caskroom.rb

```diff
@@ -47,14 +47,7 @@ def self.casks(config: nil)
         token = path.basename.to_s
 
         begin
-          if (tap_path = CaskLoader.tap_paths(token).first)
-            CaskLoader::FromTapPathLoader.new(tap_path).load(config: config)
-          elsif (caskroom_path = Pathname.glob(path.join(".metadata/*/*/*/*.rb")).first) &&
-                (!Homebrew::EnvConfig.install_from_api? || !Homebrew::API::CaskSource.available?(token))
-            CaskLoader::FromPathLoader.new(caskroom_path).load(config: config)
-          else
-            CaskLoader.load(token, config: config)
-          end
+          CaskLoader.load(token, config: config)
```

Comment:
Can someone from @Homebrew/cask confirm that this is an acceptable change?

As far as I can tell, `CaskLoader::load` will just try to load in the same order specified here. Is there another reason we needed special handling here?

Reply:
I have no idea, since I don't often modify cask code in brew. But it seems like these are legacy paths?

Reply:
Yeah, that's kind of what I figured, but the bodies of the `if` and `elsif` statements are already handled in `CaskLoader::load` in the same order (the `tap_paths` is done [here](https://github.com/Homebrew/brew/blob/9416032ee5d253fc00419f111373af54e1c6bc5d/Library/Homebrew/cask/cask_loader.rb#L228-L240) and the metadata path is handled [here](https://github.com/Homebrew/brew/blob/9416032ee5d253fc00419f111373af54e1c6bc5d/Library/Homebrew/cask/cask_loader.rb#L246-L247)), so I can't figure out why `CaskLoader::load` would return anything different from what this is doing.

Reply:
Might just have been forgotten.

Reply:
Actually, upon further inspection, there is one difference which is that `CaskLoader::load` will raise an error if the name token appears in multiple taps (aside from `homebrew/cask`) while this method will just choose the first tap which isn't necessarily the main tap.

I think it would make more sense to completely mimic `CaskLoader::load` except that when there is ambiguity, handle it better than just erroring.